ddl/ddl_api.go

```diff
@@ -7617,10 +7617,12 @@ func (d *ddl) CreateResourceGroup(ctx sessionctx.Context, stmt *ast.CreateResour
 			return err
 		}
 	}
-	if !stmt.IfNotExists {
-		if _, ok := d.GetInfoSchemaWithInterceptor(ctx).ResourceGroupByName(groupName); ok {
-			return infoschema.ErrResourceGroupExists.GenWithStackByArgs(groupName)
+
+	if _, ok := d.GetInfoSchemaWithInterceptor(ctx).ResourceGroupByName(groupName); ok {
+		if stmt.IfNotExists {
+			return nil
```

Comment:
```suggestion
					err = infoschema.ErrResourceGroupExists.GenWithStackByArgs(groupName)
		ctx.GetSessionVars().StmtCtx.AppendNote(err)
		return nil
```

Reply:
and we also needs to handle ifNotExsits clause for AlterResourceGroup().

Reply:
done